### PR TITLE
Components: Remove unused `UpgradeNudgeExpanded` component and revise `UpgradeNudge` README

### LINKED ISSUE #26745: Components: Remove unused `UpgradeNudgeExpanded` component
Currently we have the following related to upgrade nudges:

1. `UpgradeNudge` component in `client/my-sites/upgrade-nudge/`, used throughout the WordPress.com admin.

2. `UpgradeNudgeExpanded` component in `client/blocks/upgrade-nudge-expanded` that's no longer used (it tested poorly and it's use was removed in #8403, but the component remains)

3. In `client/devdocs/design/blocks`, the `UpgradeNudge` component is displayed, but it links to the readme for `UpgradeNudgeExpanded` from `client/blocks/upgrade-nudge-expanded`.

Suggested actions:

1. Re-purpose  the `UpgradeNudgeExpanded` readme for the `UpgradeNudge` component that's actually in use.
    a. Revise to apply to `UpgradeNudge`
    b. Move readme to `client/my-sites/upgrade-nudge/`
1. Delete the unused `UpgradeNudgeExpanded` component and related files.

### PR DESCRIPTION
**Pull request by mirrory**
In this branch, which fixes #26745 :

* `client/blocks/upgrade-nudge-expanded/` and all files it contained (`index.jsx`, `style.scss`, and `README.md`) have been removed.
* The `README.md` previously contained within `client/blocks/upgrade-nudge-expanded/` has been moved to `client/my-sites/upgrade-nudge/` and revised to apply to `UpgradeNudge`. I welcome any feedback and corrections on the write-up. The readmeFilePath is also set to the new file in `client/devdocs/design/blocks.jsx`.
* There were also two CSS selector references to `upgrade-nudge-expanded` in other files that have been removed. They were in `assets/stylesheets/_components.scss` (an import statement) and `client/my-sites/media-library/style.scss` (a mention of the class selector). I also did a grep search for `upgrade-nudge-expanded` throughout the project, and there are no other instances.

To test, visit:
http://calypso.localhost:3000/devdocs/blocks/upgrade-nudge
The new README is shown.

### CODE REVIEW

**Comment by mirrory** [comment]
This branch fixes issue #26745  .

**Comment by mirrory** [comment]
@creativecoder 
This PR is ready for review. Looks like I don't have permissions to put labels on it or add reviewers. But, it is done. :smile:

**Comment by creativecoder** [comment]
@mirrory Thanks so much! I've tagged with the "Needs Review" status. I'll try to get to this when I can, later this week--or (hopefully) someone else will beat me too it 🙂

**Comment by bluefuton** [comment]
Thanks for this @mirrory! We appreciate the contribution.

Would it be possible to give your branch a rebase so we can take a fresh look at it?

File: client/my-sites/upgrade-nudge/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Upgrade Nudge
```
Comment (bluefuton) [inline client/my-sites/upgrade-nudge/README.md]: Thanks for documenting this. 👏

**Comment by bluefuton** [comment]
I see that the `UpgradeNudgeExpanded` component is now back in use in `my-sites/earn/ads/wrapper.jsx`, so I'm going to close this one.

Thanks for your work on this @mirrory! I'm going to rescue the README and create a new PR with that.

**Pull request closed without merging**